fitbenchmarking/fitbenchmarking
# Updated minimum supported version of python to 3.9.x

## Description of work

Fixes #1332 

Updates the minimum supported version of python to 3.9.+. Also updates the typing to use the built in types.




## Testing Instructions



1. The GitHub actions for the default tests will now run on the minimum supported version of python (`3.9.x`). The test should pass. 
2. The GitHub actions for the `PYPI install` tests will now run on the minimum supported version of python (`3.9.x`). The test should pass. Check status [here](https://github.com/fitbenchmarking/fitbenchmarking/actions/runs/11132363990).

## Checklist
